libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,262 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/file_path_to_parser.hpp>
+#include <vast/detail/posix.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <caf/error.hpp>
+
+#include <fcntl.h>
+#include <filesystem>
+#include <memory>
+#include <string_view>
+#include <unistd.h>
+
+namespace vast::plugins::file {
+
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+  static constexpr inline auto stdin_path = "-";
+
+  auto
+  make_loader(std::span<std::string const> args, operator_control_plane&) const
+    -> caf::expected<generator<chunk_ptr>> override {
+    auto read_timeout = read_timeout_;
+    auto path = std::string{};
+    auto file_type = std::filesystem::file_type::regular;
+    auto following = false;
+    for (auto i = size_t{0}; i < args.size(); ++i) {
+      const auto& arg = args[i];
+      VAST_TRACE("processing loader argument {}", arg);
+      if (arg == "--timeout") {
+        if (i + 1 == args.size()) {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("missing duration value"));
+        }
+        if (auto parsed_duration = to<vast::duration>(args[i + 1])) {
+          read_timeout = std::chrono::duration_cast<std::chrono::milliseconds>(
+            *parsed_duration);
+          ++i;
+        } else {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("could not parse duration: {}",
+                                             args[i + 1]));
+        }
+      } else if (arg == "--uds") {
+        file_type = std::filesystem::file_type::socket;
+      } else if (arg == "-" || arg == "stdin") {
```

Comment:
What do you think about dropping the `stdin` completely here? Most tools I know only handle `-`.

Reply:
It would create a stronger semantic bridge from `from file` to `from stdin`.
But ultimately I have no strong opinion on this.

Reply:
:+1: to removing it here, if we're referring to a file it seems rather unusual. Also, since plugins names are case-insensitive this would imply that `from STDIN` works, but `from file STDIN` does not.

Reply:
I removed "stdin" as `file` argument in https://github.com/tenzir/vast/pull/3088/commits/26c17b32e7ba53a27f5027a6ab918f74b40d215c